plugins/single_plugins/move.cpp

```diff
@@ -320,6 +334,15 @@ class wayfire_move : public wf::plugin_interface_t
         workspace_may_changed.to   = output->workspace->get_current_workspace();
         workspace_may_changed.old_viewport_invalid = false;
         output->emit_signal("view-change-viewport", &workspace_may_changed);
+        /**
+         * Restore the fullscreen state so the window is
+         * not in a weird position afterwards, fullscreen
+         * are by spec the geometry of the output (or span outputs)
+         */
+        if (this->view->fullscreen)
```

Comment:
This should be done by the move_snap_helper_t::handle_input_released function.
Also, you should take into account which workspace the mouse position is on: for example if dragging in Expo, and mouse is on the workspace 3,3 and a fullscreen window is released there, then the fullscreen request should be for that workspace.

In case you don't know how to get the workspace where the mouse is(in pseudocode): `floor(last_grabbing_position.x / resolution.width) + current_workspace.x)`

Reply:
I'm not sure I understand:
I think you mean : 
```
        if (this->view->fullscreen)
        {
            view->fullscreen_request(output, true);
        }
```
to move-snap-helper.hpp
```
    virtual void handle_input_released()
    {
        for (auto v : enum_views(view))
        {
            end_wobbly(v);
        }
        if (this->view->fullscreen)
        {
            view->fullscreen_request(output, true);
        }
        view->disconnect_signal("geometry-changed", &view_geometry_changed);
    }

```
>Also, you should take into account which workspace the mouse position is on: for example if dragging in Expo, and ??>mouse is on the workspace 3,3 and a fullscreen window is released there, then the fullscreen request should be for that >workspace.

You mean inside the
```
        if (this->view->fullscreen)
        {
            view->fullscreen_request(output, true);
        }
```
do something like 
ws = floor(last_grabbing_position.x / resolution.width) + current_workspace.x)
 view->fullscreen_request(output, true, ws);

Do I get that correct?

Reply:
yes, but you should put that after disconnecting from the signal